# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/registry_dispatcher.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 2
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/registry_dispatcher.cc
--- a/security/sandbox/chromium/sandbox/win/src/registry_dispatcher.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/registry_dispatcher.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -22,7 +22,8 @@
 namespace {
 
 // Builds a path using the root directory and the name.
-bool GetCompletePath(HANDLE root, const base::string16& name,
+bool GetCompletePath(HANDLE root,
+                     const base::string16& name,
                      base::string16* complete_name) {
   if (root) {
     if (!sandbox::GetPathFromHandle(root, complete_name))
@@ -37,7 +38,7 @@
   return true;
 }
 
-}
+}  // namespace
 
 namespace sandbox {
 
@@ -45,11 +46,7 @@
     : policy_base_(policy_base) {
   static const IPCCall create_params = {
       {IPC_NTCREATEKEY_TAG,
-       {WCHAR_TYPE,
-        UINT32_TYPE,
-        VOIDPTR_TYPE,
-        UINT32_TYPE,
-        UINT32_TYPE,
+       {WCHAR_TYPE, UINT32_TYPE, VOIDPTR_TYPE, UINT32_TYPE, UINT32_TYPE,
         UINT32_TYPE}},
       reinterpret_cast<CallbackGeneric>(&RegistryDispatcher::NtCreateKey)};
 
@@ -89,7 +86,7 @@
   // it valid in this process.
   if (root) {
     if (!::DuplicateHandle(ipc->client_info->process, root,
-                           ::GetCurrentProcess(), &root, 0, FALSE,
+                           ::GetCurrentProcess(), &root, 0, false,
                            DUPLICATE_SAME_ACCESS))
       return false;
 
@@ -104,16 +101,15 @@
   params[OpenKey::NAME] = ParamPickerMake(regname);
   params[OpenKey::ACCESS] = ParamPickerMake(desired_access);
 
-  EvalResult result = policy_base_->EvalPolicy(IPC_NTCREATEKEY_TAG,
-                                               params.GetBase());
+  EvalResult result =
+      policy_base_->EvalPolicy(IPC_NTCREATEKEY_TAG, params.GetBase());
 
   HANDLE handle;
   NTSTATUS nt_status;
   ULONG disposition = 0;
-  if (!RegistryPolicy::CreateKeyAction(result, *ipc->client_info, *name,
-                                       attributes, root, desired_access,
-                                       title_index, create_options, &handle,
-                                       &nt_status, &disposition)) {
+  if (!RegistryPolicy::CreateKeyAction(
+          result, *ipc->client_info, *name, attributes, root, desired_access,
+          title_index, create_options, &handle, &nt_status, &disposition)) {
     ipc->return_info.nt_status = STATUS_ACCESS_DENIED;
     return true;
   }
@@ -137,10 +133,10 @@
   // it valid in this process.
   if (root) {
     if (!::DuplicateHandle(ipc->client_info->process, root,
-                           ::GetCurrentProcess(), &root, 0, FALSE,
+                           ::GetCurrentProcess(), &root, 0, false,
                            DUPLICATE_SAME_ACCESS))
       return false;
-      root_handle.Set(root);
+    root_handle.Set(root);
   }
 
   if (!GetCompletePath(root, *name, &real_path))
@@ -151,8 +147,8 @@
   params[OpenKey::NAME] = ParamPickerMake(regname);
   params[OpenKey::ACCESS] = ParamPickerMake(desired_access);
 
-  EvalResult result = policy_base_->EvalPolicy(IPC_NTOPENKEY_TAG,
-                                               params.GetBase());
+  EvalResult result =
+      policy_base_->EvalPolicy(IPC_NTOPENKEY_TAG, params.GetBase());
   HANDLE handle;
   NTSTATUS nt_status;
   if (!RegistryPolicy::OpenKeyAction(result, *ipc->client_info, *name,